packager/media/formats/mp2t/ts_segmenter_unittest.cc

```diff
@@ -319,14 +312,13 @@ TEST_F(TsSegmenterTest, FinalizeSegment) {
   EXPECT_CALL(*mock_pes_packet_generator_, Flush()).WillOnce(Return(true));
   EXPECT_CALL(*mock_pes_packet_generator_, NumberOfReadyPesPackets())
       .WillOnce(Return(0u));
-  EXPECT_CALL(*mock_ts_writer_, FinalizeSegment()).WillOnce(Return(true));
 
   segmenter.InjectPesPacketGeneratorForTesting(
       std::move(mock_pes_packet_generator_));
   EXPECT_OK(segmenter.Initialize(*stream_info));
   segmenter.InjectTsWriterForTesting(std::move(mock_ts_writer_));
-  segmenter.SetTsWriterFileOpenedForTesting(true);
-  EXPECT_OK(segmenter.FinalizeSegment(0, 100 /* arbitrary duration */));
+  segmenter.SetSegmentStartedForTesting(false);
```

Comment:
ditto. See the other comments.

Can you add a MuxerListener expection on `OnNewSegment` too to check the segment path is correct?